api/producer/countDocumentReference/src/v1/handler.py

```diff
@@ -0,0 +1,71 @@
+from enum import Enum
+from logging import Logger
+from typing import Any
+
+from aws_lambda_powertools.utilities.parser.models import APIGatewayProxyEventModel
+from lambda_pipeline.types import FrozenDict, LambdaContext, PipelineData
+from lambda_utils.logging import log_action
+
+from nrlf.core.common_steps import parse_headers
+from nrlf.core.constants import DbPrefix
+from nrlf.core.errors import assert_no_extra_params
+from nrlf.core.model import PaginatedResponse, ProducerRequestParams, key
+from nrlf.core.repository import Repository, type_filter
+from nrlf.core.transform import create_bundle_count
+from nrlf.core.validators import validate_type_system
+from nrlf.producer.fhir.r4.model import NextPageToken, RequestQuerySubject
+
+
+class LogReference(Enum):
+    COUNT_001 = "Searching for document references"
+
+
+@log_action(log_reference=LogReference.COUNT_001)
+def count_document_references(
+    data: PipelineData,
+    context: LambdaContext,
+    event: APIGatewayProxyEventModel,
+    dependencies: FrozenDict[str, Any],
+    logger: Logger,
+) -> PipelineData:
+
+    repository: Repository = dependencies["repository"]
+    request_params = ProducerRequestParams(**event.queryStringParameters or {})
+
+    assert_no_extra_params(
+        request_params=request_params, provided_params=event.queryStringParameters
+    )
+
+    nhs_number: RequestQuerySubject = request_params.nhs_number
+
+    ods_code_parts = data["ods_code_parts"]
+
+    validate_type_system(request_params.type, pointer_types=data["pointer_types"])
+
+    pointer_types = type_filter(
+        type_identifier=request_params.type,
+        pointer_types=data["pointer_types"],
+    )
+
+    count = 0
+    next_page_token: NextPageToken = None
+    while True:
+        response: PaginatedResponse = repository.query_gsi_2(
+            pk=key(DbPrefix.Organization, *ods_code_parts),
+            type=pointer_types,
+            nhs_number=nhs_number,
```

Comment:
Will be worth putting a custom limit here to avoid this process circling around the default page limit of 20

limit=1000 

I think 1000 is the currently highest amount but it could save us going out and in since we just want the count

Reply:
Its more to do with the payload size actually - so you can make that number as big as you want and it will stop at 1mb size

Reply:
So I upped it to 1000, but this made the testing requirement become at least 2001 records, which made the feature tests significantly slower.  Therefore I opted for a compromise and upped it to 100, which was larger but still performant during CI/CD.